modules/mono/mono_gd/gd_mono.cpp

```diff
@@ -385,6 +386,16 @@ static bool _on_core_api_assembly_loaded() {
 void GDMono::initialize() {
 	print_verbose(".NET: Initializing module...");
 
+#ifdef TOOLS_ENABLED
+	// Exit early if we're going to be generating glue code, so we can bootstrap the bindings without errors
+	List<String> cmdline_args = OS::get_singleton()->get_cmdline_args();
+	BindingsGenerator::process_cmdline(cmdline_args);
```

Comment:
I would maybe move this to an earlier point in `main`, instead of here as it doesn't feel like it belongs here and more like a trap waiting to happen when additional command line changes are done

Reply:
Are you thinking right before the `ScriptServer::init_languages();` call in main.cpp, a few lines above the `generate_bindings` call?

Reply:
TBH I didn't really think about where to best put this, just that this isn't it.
above init is probably fine.

I tried just moving the command line handling above the init call and if I add a theme db initialize to the bindings generator this appears to work just fine as well (the comment stating that the script server has to be initialized appears to be wrong)
```diff
diff --git a/main/main.cpp b/main/main.cpp
index 5e0187cc7fd..758711f1ebd 100644
--- a/main/main.cpp
+++ b/main/main.cpp
@@ -2406,20 +2406,19 @@ Error Main::setup2() {
 	initialize_navigation_server();
 	register_server_singletons();
 
-	// This loads global classes, so it must happen before custom loaders and savers are registered
-	ScriptServer::init_languages();
-
-	theme_db->initialize_theme();
-	audio_server->load_default_bus_layout();
 
 #if defined(MODULE_MONO_ENABLED) && defined(TOOLS_ENABLED)
 	// Hacky to have it here, but we don't have good facility yet to let modules
-	// register command line options to call at the right time. This needs to happen
-	// after init'ing the ScriptServer, but also after init'ing the ThemeDB,
-	// for the C# docs generation in the bindings.
+	// register command line options to call at the right time.
 	List<String> cmdline_args = OS::get_singleton()->get_cmdline_args();
 	BindingsGenerator::handle_cmdline_args(cmdline_args);
 #endif
+
+	// This loads global classes, so it must happen before custom loaders and savers are registered
+	ScriptServer::init_languages();
+	theme_db->initialize_theme();
+
+	audio_server->load_default_bus_layout();
 
 	if (use_debug_profiler && EngineDebugger::is_active()) {
 		// Start the "scripts" profiler, used in local debugging.
diff --git a/modules/mono/editor/bindings_generator.cpp b/modules/mono/editor/bindings_generator.cpp
index 8504fb2ac6a..2d77e5d2f76 100644
--- a/modules/mono/editor/bindings_generator.cpp
+++ b/modules/mono/editor/bindings_generator.cpp
@@ -39,6 +39,7 @@
 #include "core/io/file_access.h"
 #include "core/os/os.h"
 #include "main/main.h"
+#include "scene/theme/theme_db.h"
 
 #include "../godotsharp_defs.h"
 #include "../utils/naming_utils.h"
@@ -3967,6 +3968,7 @@ void BindingsGenerator::handle_cmdline_args(const List<String> &p_cmdline_args)
 	}
 
 	if (glue_dir_path.length()) {
+		ThemeDB::get_singleton()->initialize_theme();
 		handle_cmdline_options(glue_dir_path);
 		// Exit once done
 		cleanup_and_exit_godot();
```

The only diff that is causes in the generated code is the value of the `NativeCalls::godot_api_hash` of godotsharp but that files is actually unused anyways to this doesn't change anything.

Reply:
Indeed, the reason this was moved to main is to prevent a crash after some theme related changes were introduced: #74565. This PR effectively reverses that patch, so I would expect it to re-introduce that issue.

I can't review @RedworkDE's patch as I'm that is not my area, and it's easy to introduce a regression. I think it would be better to just skip C# initialization if the glue generation command line argument is detected:

```diff
void CSharpLanguage::init() {
+#ifdef TOOLS_ENABLED
+	if (OS::get_singleton()->get_cmdline_args().find("--generate-mono-glue")) {
+		return;
+	}
+#endif
```

Reply:
I moved things around a bit to try and make it cleaner without changing the existing initialization order (and hopefully not introduce regressions), see if it looks better. I would rather not hardcode the flag if I can help it, it makes the code more fragile - hopefully this latest refactor feels better.

Reply:
I feel like **neikeq**'s suggestion would be enough and simpler than all the changes this PR is currently making. I don't think there's an issue with hardcoding the flag, we can keep it in a constant/static field if you prefer like in [`BindingsGenerator::generate_all_glue_option`](https://github.com/godotengine/godot/blob/964a5353dbf5fc0dca4c4b7b9cfa5631dd5ab568/modules/mono/editor/bindings_generator.cpp#L3921).